### PR TITLE
[pls review] Fix contentful image handling

### PR DESCRIPTION
**Pull request by axe312ger**
* Export the schemes for #4205 
* Allow background image setting for proper output when cropping
* Return correct aspect ratio for responsive image nodes

### CODE REVIEW

**Comment by axe312ger** [comment]
@Khaledgarbaya maybe u can have a look :)

File: packages/gatsby-source-contentful/src/__tests__/extend-node-type.js [hunk 1]
```diff
@@ -94,8 +95,9 @@ describe(`contentful extend node type`, () => {
     it(`generates responsive sizes data for images using all options`, async () => {
       const resp = await resolveResponsiveSizes(image, {
         maxWidth: 450,
-        maxHeight: 399,
+        maxHeight: 400,
```
Comment (KyleAMathews) [inline packages/gatsby-source-contentful/src/__tests__/extend-node-type.js]: Could you change this back to 399? It was set this way to test that rounding works I believe.

File: packages/gatsby-source-contentful/src/__tests__/extend-node-type.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ describe(`contentful extend node type`, () => {
         width: 450,
         height: 399,
         quality: 50,
+        background: `rgb:000000`,
```
Comment (KyleAMathews) [inline packages/gatsby-source-contentful/src/__tests__/extend-node-type.js]: What's this for?

File: packages/gatsby-source-contentful/src/__tests__/extend-node-type.js (hunk 1 above)
Comment (axe312ger) [reply packages/gatsby-source-contentful/src/__tests__/extend-node-type.js]: Ohh, felt so weird to me to have such an non-practical height and a resulting aspect ratio of `1.1278195488721805`

but yes, I can do that :)

File: packages/gatsby-source-contentful/src/__tests__/extend-node-type.js (hunk 2 above)
Comment (axe312ger) [reply packages/gatsby-source-contentful/src/__tests__/extend-node-type.js]: The background is set when doing cropping operations where the source image does not completely cover the result.

See:

https://images.contentful.com/wln3fygpfpvt/7orLdboQQowIUs22KAW4U/6c709893ae83938bfef443a7e5f8c658/matt-palmer-254999.jpg?w=1180&h=480&fit=pad


https://images.contentful.com/wln3fygpfpvt/7orLdboQQowIUs22KAW4U/6c709893ae83938bfef443a7e5f8c658/matt-palmer-254999.jpg?w=1180&h=480&fit=pad&bg=rgb:000000

**Comment by axe312ger** [comment]
@KyleAMathews this is ready for re-review :)

File: packages/gatsby-source-contentful/src/__tests__/extend-node-type.js (hunk 2 above)
Comment (KyleAMathews) [reply packages/gatsby-source-contentful/src/__tests__/extend-node-type.js]: Ah cool 👍

File: packages/gatsby-source-contentful/src/__tests__/__snapshots__/extend-node-type.js.snap [hunk 3]
```diff
@@ -70,16 +70,16 @@ Object {
 
 exports[`contentful extend node type resolveResponsiveSizes generates responsive sizes data for images using all options 1`] = `
 Object {
-  "aspectRatio": 0.75,
+  "aspectRatio": 1.1278195488721805,
```
Comment (KyleAMathews) [inline packages/gatsby-source-contentful/src/__tests__/__snapshots__/extend-node-type.js.snap]: Looks like something broke this?

Comment (axe312ger) [reply packages/gatsby-source-contentful/src/__tests__/__snapshots__/extend-node-type.js.snap]: This is the outcome of the bugfix for this result `w=450&h=399` https://github.com/gatsbyjs/gatsby/pull/4207/commits/b0c658cf0d9947aff0699547b7675d66cb774515

It returned the original aspect ratio while the resulting one should be returned

Comment (KyleAMathews) [reply packages/gatsby-source-contentful/src/__tests__/__snapshots__/extend-node-type.js.snap]: ahhh :-) cool!

**Comment by KyleAMathews** [comment]
Thanks!

**Pull request merged**